File: packages/swift/Sources/CatalystSwift/Clients/FeatureFlagsClient.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

/// Client for Feature Flags API endpoints.
public final class FeatureFlagsClient: Sendable {
  private let client: CatalystSwift

  init(client: CatalystSwift) {
    self.client = client
  }

  /// Gets the feature flags enabled for the current user.
  public func me() async throws -> [String] {
    let response: CatalystFeatureFlags = try await client.request(FeatureFlagsEndpoint.me)
    return response.flags
  }
}

File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/clients/FeatureFlagsClient.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.clients

import com.natsuneko.catalyst.http.CatalystHttpClient
import com.natsuneko.catalyst.models.CatalystFeatureFlags

/**
 * Client for Feature Flags API endpoints
 */
class FeatureFlagsClient internal constructor(
    private val httpClient: CatalystHttpClient
) {
    /**
     * Gets the feature flags enabled for the current user
     */
    suspend fun getMyFeatureFlags(): List<String> =
        httpClient.get<CatalystFeatureFlags>("/feature-flags/v1/me").flags
}
